pkg/compactor/bucket_compactor.go

```diff
@@ -422,6 +429,59 @@ func (c *BucketCompactor) runCompactionJob(ctx context.Context, job *Job) (shoul
 	return true, compIDs, nil
 }
 
+// verifyCompactedBlocksTimeRanges does a full run over the compacted output blocks
+// and verifies that they satisfy the min/maxTime from the input blocks
+func verifyCompactedBlocksTimeRanges(ctx context.Context, compIDs []ulid.ULID, inputBlocksMinTime, inputBlocksMaxTime int64, blockSyncConcurrency int, subDir string) error {
+	inputBlocksMinTimeFound := atomic.NewBool(false)
+	inputBlocksMaxTimeFound := atomic.NewBool(false)
+
+	err := concurrency.ForEachJob(ctx, len(compIDs), blockSyncConcurrency, func(ctx context.Context, idx int) error {
```

Comment:
Why do we do this concurrently? I think it complicates the logic and code here (eg. need to use atomics), and is not necessary.